libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -754,6 +754,53 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		}
 		break;
 	}
+	case FunctionType::Kind::Creation:
+	{
+		solAssert(!functionType->gasSet(), "Gas limit set for contract creation.");
+		solAssert(
+			functionType->returnParameterTypes().size() == 1,
+			"Constructor should return only one type"
+		);
+
+		TypePointers argumentTypes;
+		string constructorParams;
+		for (auto const& arg: arguments)
+		{
+			argumentTypes.push_back(arg->annotation().type);
+			constructorParams += (constructorParams.empty() ? "" : ", ") + IRVariable{*arg}.commaSeparatedList();
+		}
+
+		ContractDefinition const* contract =
+			&dynamic_cast<ContractType const&>(*functionType->returnParameterTypes().front()).contractDefinition();
+
+		Whiskers t(R"(
+			let size := datasize("<object>")
+			let memPos := <allocate>(size)
```

Comment:
Small optimization: You do not need to allocate here because we do not need the memory area after the creation call and `abiEncode` does not allocate memory of its own.

Reply:
What I really want is free memory pointer. I can achieve it with allocate(0)?

Reply:
This will give freeMemoryPointer, but will not update freepointer itself, so I guess it is good like this. Other option is to use freeMemoryPointer directly with mload